### ISSUE TITLE
STL-10 Testing Protocol

### ISSUE THREAD

**Issue by gaa-cifasis** [pytorch/vision#167]
Class STL10 does not support recommended testing protocol. 
See STL-10 official page: https://cs.stanford.edu/~acoates/stl10/

**Comment by dlmacedo** [comment]
This is important!

**Comment by alykhantejani** [comment]
Hi @gaa-cifasis sorry for the delayed response. I'm not too familiar with this dataset, can you elaborate a bit on what is wrong?

As far as I can tell, the `STL10` Dataset is [loading the test set correctly](https://github.com/pytorch/vision/blob/master/torchvision/datasets/stl10.py#L40). As for the testing protocol, I would assume that this would be handled and by the code consuming the data and not the dataset itself?

**Comment by gaa-cifasis** [comment]
I think this issue is related with the loading of the training set. According to the website:

> The file fold_indices.txt contains the (zero-based) indices of the examples to be used for each training fold

but STL10 just loads all the folds for training. Therefore, the protocol is not followed and the results cannot be easily compared with the other papers.

**Comment by omair-kg** [comment]
For this you could use the subsetrandomsampler.

```
folds = np.loadtxt('fold_indices.txt')

fold = folds[use_fold]
fold = torch.from_numpy(fold)

dataloader = torch.utils.data.DataLoader(dataset, batch_size=batchSize, sampler=SubsetRandomSampler(fold), num_workers=2,pin_memory=False)
```
That should do it

**Comment by fmassa** [comment]
While we can use the subset approach from @omair-kg , I believe we should add the folds as as option for STL-10.
